sotodlib/io/bookbinder2.py

```diff
@@ -0,0 +1,1118 @@
+        if 'Corrected_Corotation' in block.data:
+            corotation = np.interp(times, block.times, block.data['Corrected_Corotation'])
+
+        anc_frame_data = []
+        for oframe_idx in np.unique(frame_idxs):
+            if file_idxs[oframe_idx] != cur_file_idx:
+                close_writer(writer)
+                cur_file_idx = file_idxs[oframe_idx]
+                fname = os.path.join(outdir, f'A_ancil_{cur_file_idx:0>3}.g3')
+                out_files.append(fname)
+                writer = core.G3Writer(fname)
+
+            m = frame_idxs == oframe_idx
+            ts = times[m]
+
+            oframe = core.G3Frame(core.G3FrameType.Scan)
+
+            i0, i1 = np.where(m)[0][[0, -1]]
+            oframe['sample_range'] = core.G3VectorInt([int(i0), int(i1+1)])
+            oframe['book_id'] = self.book_id
+
+            anc_data = core.G3TimesampleMap()
+            anc_data.times = core.G3VectorTime(ts * core.G3Units.s)
+            if az is not None:
+                anc_data['az_enc'] = core.G3VectorDouble(az[m])
+                anc_data['el_enc'] = core.G3VectorDouble(el[m])
+            if boresight is not None:
+                anc_data['boresight_enc'] = core.G3VectorDouble(boresight[m])
+            if corotation is not None:
+                anc_data['corotation_enc'] = core.G3VectorDouble(corotation[m])
+            oframe['ancil'] = anc_data
+            writer(oframe)
+            anc_frame_data.append(anc_data)
+
+            self.add_acu_summary_info(oframe, ts[0], ts[-1])
+
+        # Save this to be added to detector files
+        self.anc_frame_data = anc_frame_data
+        self.out_files = out_files
+    
+    def add_acu_summary_info(self, frame, t0, t1):
+        """
+        Adds ACU summary information to a G3Frame. This will add the following
+        info if data is present in the HK dataset::
+            - azimuth_mode: (str)
+                Azimuth_mode, pulled from the ACU summary data. `ProgramTrack`
+                means that this frame contains scan data, and `Preset` means the
+                telescope is slewing.
+            - azimuth_velocity_mean / azimuth_velocity_std: (float / float)
+                Mean and standard deviation of the az velocity (deg / sec)
+            - elevation_velocity_mean / elevaction_velocity_std: (float / float)
+                Mean and standard deviation of the el velocity (deg / sec)
+            
+        Params
+        ----------
+        frame : G3Frame
+            Frame to add data to
+        t0 : float
+            Start time of frame (unix time)
+        t1 : float
+            Stop time of frame (unix time)
+        """
+        bl = self.blocks.get('ACU_summary_output')
+        if bl.data:
+            az_mode = bl.data["Azimuth_mode"]
+            acu_summary_times = bl.times
+            m = (t0 < acu_summary_times) & (acu_summary_times <= t1)
+            if not np.any(m):
+                frame['azimuth_mode'] = 'None'
```

Comment:
Is this equivalent to a stopped state?

Reply:
I think it's good to keep a distinction between having no `az` data, and being stopped.

Reply:
Ah whoops I misread the if condition. What’s marker for a stopped state? when both `azimuth_velocity_mean` and `elevation_velocity_mean` are within atol?

Reply:
There are more detailed az-states such as "Stop" / more in the HK books. I decided not to include them in this summary since based on our conversation it seemed like we really only cared about "Scanning" vs "Not-Scanning" (or slewing). Unless there's a reason to go into more depth here, I don't think we really need to, since that involves making decisions about what the 'summary' should be when there's a combination of different modes within a frame.

As it is now, the azimuth_mode summary is:
- 'None' --> No az-mode data in the frame
- 'Program Track' --> There exists at least one sample where it's scanning
- 'Preset' --> Not scanning